prog/analysis.go

```diff
@@ -309,3 +310,98 @@ func encodeFallbackSignal(typ, id, aux int) uint32 {
 func decodeFallbackSignal(s uint32) (typ, id, aux int) {
 	return int(s & 7), int((s >> 3) & fallbackCallMask), int(s >> 16)
 }
+
+type pair struct {
+	addr uint64
+	len  uint64
+}
+
+func detectIntersection(ranges []pair) bool {
+	sort.SliceStable(ranges, func(i, j int) bool {
+		return ranges[i].addr < ranges[j].addr
+	})
+	for i := 0; i < len(ranges)-1; i++ {
+		if ranges[i+1].addr < ranges[i].addr+ranges[i].len {
+			return true
+		}
+	}
+	return false
+}
+
+func max(a, b uint64) uint64 {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func filterArguments(call *Call, pointers map[Arg]uint64, p *Prog) bool {
+	var ranges []pair
+	ForeachArg(call, func(arg Arg, ctx *ArgCtx) {
+		switch a := arg.(type) {
+		case *PointerArg:
+			if ctx.Base == nil {
+				return
+			}
+			value, ok := pointers[arg]
+			addr := p.Target.PhysicalAddr(ctx.Base) + ctx.Offset
+			addr -= arg.Type().UnitOffset()
+			if ok {
+				ranges = append(ranges, pair{addr, value})
+			} else if a.Res != nil {
+				ranges = append(ranges, pair{addr, max(a.Res.Size(), 1)})
+			}
+		case *GroupArg:
+			addr := p.Target.PhysicalAddr(ctx.Base) + ctx.Offset
+			addr -= arg.Type().UnitOffset()
+			ranges = append(ranges, pair{addr, max(arg.Size(), 1)})
+		}
+
```

Comment:
This newline doesn't seem necessary (unless gofmt adds it back when you delete it)